### PR TITLE
Fixed nokogiri project

### PR DESCRIPTION
**Pull request by fuzzy-boiii23a**
Fixed introspector, improved checking of pointers, and added memory sanitizer

### CODE REVIEW

File: projects/nokogiri/build.sh [hunk 1]
```diff
@@ -57,10 +57,19 @@ then
...
+then
+  export SANITIZER_OPTS="-fno-omit-frame-pointer -gline-tables-only -fuse-ld=gold -flto -Wno-unused-command-line-argument"
+  export SANITIZER_LINK=""
+fi
 
-
-export CXXFLAGS="-O0 $CXXFLAGS $SANITIZER_OPTS"
-export CFLAGS="-O0 $CFLAGS $SANITIZER_OPTS"
+export CXXFLAGS="-O0 -fno-inline-functions $CXXFLAGS $SANITIZER_OPTS"
+export CFLAGS="-O0 -fno-inline-functions  $CFLAGS $SANITIZER_OPTS"
```
Comment (DavidKorczynski) [inline projects/nokogiri/build.sh:72]: Am not sure I understand, what is the point of setting all these environment variables? These are set by the OSS-Fuzz infra itself, why is there a need to redo it?

**Review by DavidKorczynski — CHANGES_REQUESTED** [review]

File: projects/nokogiri/build.sh (hunk 1 above)
Comment (fuzzy-boiii23a) [reply projects/nokogiri/build.sh:72]: I'll give it a try without defining any of them myself, I just thought if I did them myself I didn't have to worry about external factors changing such as predefined environment variables, my bad

Comment (fuzzy-boiii23a) [reply projects/nokogiri/build.sh]: alrighty made those changes @DavidKorczynski thanks for the suggestions! I'll make sure to not make that mistake in future integrations

**Review by DavidKorczynski — CHANGES_REQUESTED** [review]
What's the point of the `ENGINE_LINK` vars? Why not use `LIB_FUZZING_ENGINE` provided by OSS-Fuzz infra?

**Comment by fuzzy-boiii23a** [comment]
> What's the point of the `ENGINE_LINK` vars? Why not use `LIB_FUZZING_ENGINE` provided by OSS-Fuzz infra?

Fixed this just then and tested and everything working as expected locally

**Review by DavidKorczynski — CHANGES_REQUESTED** [review]
The same comments as I had above applies to the remaining environment variable being set.

**Comment by fuzzy-boiii23a** [comment]
> The same comments as I had above applies to the remaining environment variable being set.

Done, sorry should be all good now as I removed them and everything is still working as expected, thanks for the help :)

**Review by DavidKorczynski — APPROVED** [review]
Thanks for addressing the reviews! Looks much better now.

**Pull request merged**